iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/client/HiveRESTCatalogClient.java

```diff
@@ -0,0 +1,244 @@
+    Pattern pattern = Pattern.compile(regex);
+
+    return restCatalog.listNamespaces(Namespace.empty()).stream()
+        .map(Namespace::toString)
+        .filter(pattern.asPredicate())
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public List<String> getAllDatabases(String catName) {
+    return getDatabases(catName, "*");
+  }
+
+  @Override
+  public List<String> getTables(String catName, String dbName, String tablePattern) {
+    validateCurrentCatalog(catName);
+
+    // Convert the Hive glob pattern to a Java regex.
+    String regex = tablePattern.replace("*", ".*");
+    Pattern pattern = Pattern.compile(regex);
+
+    // List tables from the specific database (namespace) and filter them.
+    return restCatalog.listTables(Namespace.of(dbName)).stream()
+        .map(TableIdentifier::name)
+        .filter(pattern.asPredicate())
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public List<String> getAllTables(String catName, String dbName) {
+    return getTables(catName, dbName, "*");
+  }
+
+  @Override
+  public void dropTable(Table table, boolean deleteData, boolean ignoreUnknownTab, boolean ifPurge) throws TException {
+    restCatalog.dropTable(TableIdentifier.of(table.getDbName(), table.getTableName()));
+  }
+
+  private void validateCurrentCatalog(String catName) {
+    if (!restCatalog.name().equals(catName)) {
+      throw new IllegalArgumentException(
+          String.format("Catalog name '%s' does not match the current catalog '%s'", catName, restCatalog.name()));
+    }
+  }
+
+  @Override
+  public boolean tableExists(String catName, String dbName, String tableName) {
+    validateCurrentCatalog(catName);
+    return restCatalog.tableExists(TableIdentifier.of(dbName, tableName));
+  }
+
+  @Override
+  public Database getDatabase(String catalogName, String databaseName) {
+    validateCurrentCatalog(catalogName);
+    return restCatalog.listNamespaces(Namespace.empty()).stream()
+        .filter(namespace -> namespace.levels()[0].equals(databaseName))
+        .map(namespace -> {
+          Database database = new Database();
+          database.setName(String.join(NAMESPACE_SEPARATOR, namespace.levels()));
+          Map<String, String> namespaceMetadata = restCatalog.loadNamespaceMetadata(Namespace.of(databaseName));
+          database.setLocationUri(namespaceMetadata.get(CatalogUtils.LOCATION));
+          database.setCatalogName(restCatalog.name());
+          database.setOwnerName(namespaceMetadata.get(DB_OWNER));
+          try {
+            database.setOwnerType(PrincipalType.valueOf(namespaceMetadata.get(DB_OWNER_TYPE)));
+          } catch (Exception e) {
+            LOG.warn("Can not set ownerType: {}", namespaceMetadata.get(DB_OWNER_TYPE), e);
+          }
+          return database;
+        }).findFirst().get();
+  }
+
+  @Override
+  public Table getTable(GetTableRequest getTableRequest) throws TException {
+    org.apache.iceberg.Table icebergTable;
```

Comment:
should we validate the catalog here as well?

Reply:
Done